numpy/ma/core.py

```diff
@@ -3802,7 +3802,10 @@ def __iadd__(self, other):
         else:
             if m is not nomask:
                 self._mask += m
-        ndarray.__iadd__(self._data, np.where(self._mask, 0, getdata(other)))
+        ndarray.__iadd__(
```

Comment:
Some other things to try. First, I think you can do
```
self._data += ...
```
Another posibility is to use the `np.add` ufunc with the `out` argument, which allows you to set `casting='unsafe'`. The difference of doing this directly as opposed to ndarray.__iadd__, is that the ndarray functions can be set dynamically after loading the module. However, we set them to ufuncs on umath load, and I don't know of anyone else who uses that feature.